setup.Rmd

```diff
@@ -13,52 +13,36 @@ install.packages(c("devtools", "roxygen2", "testthat", "knitr"))
-    -   Most readers can use the free, open source version of RStudio Desktop.
+Download the current version of RStudio Desktop here: <https://posit.co/download/rstudio-desktop/>.
+Most readers can use the free, open source version of RStudio Desktop.
 
 ## devtools, usethis, and you {#sec-setup-usage}
 
 > "I am large, I contain multitudes."
 >
 > --- Walt Whitman, Song of Myself
 
-After 7 years of development, devtools had grown into a rather unwieldy package, making maintenance difficult.
-Version 2.0.0, released in late 2018, marked the [conscious uncoupling](https://www.tidyverse.org/articles/2018/10/devtools-2-0-0/) of devtools, with most functionality moving into seven smaller packages.
-Through various means, devtools continues to expose all its usual functionality, although it is mostly maintained elsewhere.
+As mentioned in @sec-intro-phil, devtools is a 'meta-package', encompassing and exposing functionality maintained in seven smaller packages[^setup-1].
 For example, devtools might provide a wrapper function in order to set user-friendly defaults, introduce helpful interactive behaviour, or to combine functionality from multiple sub-packages.
+In some cases it simply re-exports a function from another package to make it easily available when devtools is loaded.
 
-What's our recommended approach to devtools and its constituent packages?
-It varies, depending on whether you're working in useR or developeR mode:
-
--   For interactive use, useRs should attach devtools and think of it as the provider of your favorite functions for package development.
--   For programmatic use, such as inside another package, developeRs should NOT depend on devtools, but should instead access functions via the package that is their primary home.
-    -   devtools should rarely appear in the role of `pkg` in a qualified call of the form `pkg::fcn()`. Instead, `pkg` should be the package where `fcn()` is defined.
-    -   An exception to this is that we continue to feature `devtools::install_github()` as the way to install the development version of a package in its README, even though `install_github()` actually lives in the remotes package. That's because this piece of advice pertains to interactive use, where we prefer to emphasize devtools.
--   Try to report bugs on the package that is a function's primary home. The help for `devtools::fcn()` usually states when devtools is re-exporting a function from another package.
-
-Example of how to simulate installing and loading a package, during interactive development:
-
-```{r eval = FALSE}
-library(devtools)
-load_all()
-```
+[^setup-1]: [remotes](https://remotes.r-lib.org/), [pkgbuild](https://github.com/r-lib/pkgbuild), [pkgload](https://github.com/r-lib/pkgload), [rcmdcheck](https://github.com/r-lib/rcmdcheck), [revdepcheck](https://github.com/r-lib/revdepcheck), [sessioninfo](https://github.com/r-lib/sessioninfo), [usethis](https://usethis.r-lib.org/)
```

Comment:
```suggestion
[^setup-1]: At the time of writing, devtools exposes functionality from [remotes](https://remotes.r-lib.org/), [pkgbuild](https://github.com/r-lib/pkgbuild), [pkgload](https://github.com/r-lib/pkgload), [rcmdcheck](https://github.com/r-lib/rcmdcheck), [revdepcheck](https://github.com/r-lib/revdepcheck), [sessioninfo](https://github.com/r-lib/sessioninfo), and [usethis](https://usethis.r-lib.org/)
```

More rewording to be a bit more future-proof. Shouldn't this include roxygen2 and testthat?

Reply:
Some of these links are to the GitHub repos, and some are to the pkgdown pages. I got the original list (and links) from the [conscious uncoupling](https://www.tidyverse.org/blog/2018/10/devtools-2-0-0/#conscious-uncoupling) blog, but that was a while ago now. Should they all be pkgdown now? On the [devtools pkgdown site](https://devtools.r-lib.org/#conscious-uncoupling) all of the links go to the repos...

Reply:
Re: why the blog didn't list roxygen2 and testhat. Presumably that's because those two packages were already well-established and had a full life of their own at the time of the uncoupling. Versus being created by carving out functionality from devtools itself.

So I think, given how we're approaching the history here, it's appropriate to include roxygen2 and testthat in this list.

I think we should link to a pkgdown site when it exists. But it won't for some of the more low-level packages, in which case GitHub repo is fine.

Reply:
Perfect, thanks for that explanation - I'll update the links to pkgdown where available.